xarray_filters/datasets.py

```diff
@@ -1,117 +1,415 @@
+              in the final step of any chained transformers    # FOR NOW Y WILL BE 1D
+    """
+    skl_argspec = inspect.getfullargspec(skl_sampler_func)
+    # Here is where we use the assumption that the make_* function from sklearn
+    # has all positional or keyword arguments, all of them with defaults; it
+    # could be easily adapted to more flexible setups. TODO: make this more
+    # robust; users of the library may apply this to functions that do not
+    # satisfy the assumptions listed above.
+    assert not skl_argspec.varargs, "{} has variable positional arguments".format(skl_sampler_func.__name__)
+    assert not skl_argspec.kwonlyargs, "{} has keyword-only arguments".format(skl_sampler_func.__name__)
+    assert len(skl_argspec.args) == len(skl_argspec.defaults), \
+            "Some args of {} have no default value".format(skl_sampler_func.__name__)
+    skl_params = [inspect.Parameter(name=pname, kind=inspect.Parameter.POSITIONAL_OR_KEYWORD, default=pdefault)
+                  for (pname, pdefault) in zip(skl_argspec.args, skl_argspec.defaults)]
+    default_astype = 'dataset'
+    def wrapper(*args, **kwargs):
+        '''
+        TODO - Gui see if we can do something
+               like this instead so that we don't
+               return an XyTransformer but
+               rather the output of an
+               XyTransformer method
+
+        sk_args, sk_kw = ...do something to get only
+                          args/kwargs that sklearn.dataset
+                          func needs
+        other_args, other_kw = ...do something to get only
+                               args/kwargs related to
+                               XyTransformer
+        Check sklearn docs, but I think that
+            there are no args - just kwargs
+        X, y = skl_sampler_func(*sk_args, **sk_kw)
+        xyt = XyTransformer(X, y)
+        method = .....some way of mapping to the correct
+                      XyTransformer method, like getting
+                      the method out of other_kw - related to "astype"
+        func = getattr(xyt, method)
+        return func(*other_args, **other_kw)
+
+        Some of other_kw -> may be calculated from sk_kw or vice versa?, e.g.
+            sk_kw - n_samples
+            other_kw - shape (must be consistent)
+            Take shape keyword or Exception if n_samples given
+
+        ---
+        All optional/custom args are keyword arguments.
+
+        '''
+        # Step 1: process positional and keyword arguments
+        skl_kwds = skl_argspec.args + skl_argspec.kwonlyargs
+        # splitting arguments into disjoint sets; astype is a special argument
+        skl_kwargs = {k: val for (k, val) in kwargs.items() if k in skl_kwds}
+        astype = kwargs.get('astype', default_astype)
+        type_kwargs = {k: val for (k, val) in kwargs.items() if (k not in
+            skl_kwds and k != 'astype')} 
+
+        # Step 2: obtain the XyTransformer object
+        try:
```

Comment:
@PeterDSteinberg 

Now I'm thinking a better option is to just read in the output of the sklearn make_* function and explicitly check if the lenght of the output is 2, then raise an error if that is the case. Yea/Nay?